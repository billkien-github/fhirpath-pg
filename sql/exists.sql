SELECT  '.a.b.exists()'::fhirpath;

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.exists()', 'decimal'); 
SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.exists()', 'decimal'); 
SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.c.exists()', 'decimal'); 
SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.d.exists()', 'decimal'); 
SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.f.exists()', 'decimal'); 

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.c.exists()', 'decimal'); 

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.e.exists()', 'decimal'); 
SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.f.e.exists()', 'decimal'); 

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.c.d.exists()', 'decimal'); 
SELECT fhirpath_exists('{"a":{"b": {"cDate": 5.3}}}', '.a.b.c.exists()', 'Polymorphic'); 

SELECT fhirpath_exists('{"aDate":{"b": {"c": 5.3}}}', '.a.b.c.exists()', 'Polymorphic'); 
SELECT fhirpath_exists('{"a":{"bDate": {"c": 5.3}}}', '.a.b.c.exists()', 'Polymorphic'); 
