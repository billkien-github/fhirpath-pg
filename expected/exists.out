SELECT  '.a.b.exists()'::fhirpath;
   fhirpath    
---------------
 .a.b.exists()
(1 row)

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 t
(1 row)

SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 t
(1 row)

SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.c.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 t
(1 row)

SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.d.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 t
(1 row)

SELECT fhirpath_exists('{"a":{"b": [{"c": 5.3}, {"d": 1}]}}', '.a.b.f.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 f
(1 row)

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.c.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 t
(1 row)

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.e.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 f
(1 row)

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.f.e.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 f
(1 row)

SELECT fhirpath_exists('{"a":{"b": {"c": 5.3}}}', '.a.b.c.d.exists()', 'decimal'); 
 fhirpath_exists 
-----------------
 f
(1 row)

SELECT fhirpath_exists('{"a":{"b": {"cDate": 5.3}}}', '.a.b.c.exists()', 'Polymorphic'); 
 fhirpath_exists 
-----------------
 t
(1 row)

SELECT fhirpath_exists('{"aDate":{"b": {"c": 5.3}}}', '.a.b.c.exists()', 'Polymorphic'); 
 fhirpath_exists 
-----------------
 f
(1 row)

SELECT fhirpath_exists('{"a":{"bDate": {"c": 5.3}}}', '.a.b.c.exists()', 'Polymorphic'); 
 fhirpath_exists 
-----------------
 f
(1 row)

